src/components/application_manager/include/application_manager/policies/policy_handler.h

```diff
@@ -112,6 +112,159 @@ class PolicyHandler : public PolicyHandlerInterface,
+   * @param app_types additional list of HMI type to search in it
+   * @return true if hmi is contained in policy or app_types
+   */
+  bool CheckHMIType(const std::string& application_id,
+                    mobile_apis::AppHMIType::eType hmi,
+                    const smart_objects::SmartObject* app_types) OVERRIDE;
+
+  /**
+   * Notifies about changing HMI level
+   * @param device_id unique identifier of device
+   * @param policy_app_id unique identifier of application in policy
+   * @param hmi_level default HMI level for this application
+   */
+  void OnUpdateHMILevel(const std::string& device_id,
+                        const std::string& policy_app_id,
+                        const std::string& hmi_level) OVERRIDE;
+
+  /**
+   * Checks access to equipment of vehicle for application by RPC
+   * @param device_id unique identifier of device
+   * @param app_id policy id application
+   * @param module type
+   * @param rpc name of rpc
+   * @param params parameters list
+   */
+  application_manager::TypeAccess CheckAccess(
+      const PTString& device_id,
+      const PTString& app_id,
+      const PTString& module,
+      const std::string& rpc,
+      const std::vector<PTString>& params) OVERRIDE;
+
+  /**
+   * Checks access to module for application
+   * @param app_id policy id application
+   * @param module
+   * @return true if module is allowed for application
+   */
+  bool CheckModule(const PTString& app_id, const PTString& module) OVERRIDE;
+
+  /**
+   * Sets access to equipment of vehicle for application by RPC
+   * @param device_id unique identifier of device
+   * @param app_id policy id application
+   * @param module type
+   * @param allowed true if access is allowed
+   */
+  void SetAccess(const PTString& device_id,
+                 const PTString& app_id,
+                 const PTString& module,
+                 bool allowed) OVERRIDE;
+
+  /**
+   * Resets access application to all resources
+   * @param device_id unique identifier of device
+   * @param app_id policy id application
+   */
+  void ResetAccess(const PTString& device_id, const PTString& app_id) OVERRIDE;
+
+  /**
+   * Resets access by group name for all applications
+   * @param module type
+   */
+  void ResetAccess(const std::string& module) OVERRIDE;
+
+  /**
+   * Sets device as primary device
+   * @param dev_id ID device
+   */
+  void SetPrimaryDevice(const PTString& dev_id) OVERRIDE;
+
+  /**
+   * Resets driver's device
+   */
+  void ResetPrimaryDevice() OVERRIDE;
+
+  /**
+   * Return id of primary device
+   */
+  uint32_t PrimaryDevice() const OVERRIDE;
+
+  /**
+   * Sets mode of remote control (on/off)
+   * @param enabled true if remote control is turned on
+   */
+  void SetRemoteControl(bool enabled) OVERRIDE;
+
+  /**
+   * @brief If remote control is enabled
+   * by User and by Policy
+   */
+  bool GetRemoteControl() const OVERRIDE;
+
+  /**
+   * @brief Notifies passengers' apps about change
```

Comment:
@yang1070
Some parts have been left to consider logic extension in future as this implementation is assumed as first part. Do you want to remove that?